.github/workflows/destroy_review_app.yml

```diff
@@ -3,72 +3,59 @@ name: Delete review
 on:
   pull_request:
     types: [closed]
-    branches: [main]
-
-permissions:
-  id-token: write
-  packages: write
+    branches: [2323-use-delete-review-app-github-action]
+  workflow_dispatch:
+    inputs:
+      pr_number:
+        description: PR number of review app to delete
+        required: true
+        type: string
 
 jobs:
   delete-review-app:
     name: Delete Review App ${{ github.event.pull_request.number }}
     concurrency: deploy_review_${{ github.event.pull_request.number }}
-    if: ${{ !contains(github.event.pull_request.labels.*.name, 'dependencies') }}
-    runs-on: ubuntu-24.04
     environment: review
+    if: contains(github.event.pull_request.labels.*.name, 'deploy') ||  ${{ github.event_name }} == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: "read"
+      id-token: "write"
     steps:
-      - uses: actions/checkout@v4
-
-      - uses: azure/login@v2
-        with:
-          client-id: ${{ secrets.AZURE_CLIENT_ID }}
-          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
-          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
-
-      - uses: hashicorp/setup-terraform@v3
-        with:
-          terraform_version: 1.5.4
-          terraform_wrapper: false
-
-      - name: Set environment variables
+      - name: Checkout
+        uses: actions/checkout@v4
+        
+      - name: set PR_NUMBER
+        id: config
         run: |
-          state_file_name=${{ github.event.pull_request.number }}.tfstate
-
-          echo "TF_STATE_FILE=$state_file_name" >> $GITHUB_ENV
-
-          state_file_status=$(az storage blob list -c terraform-state \
-            --account-name "s189t01cpdnpqrvtfsa" \
-            --prefix $state_file_name --query "[].name" -o tsv)
-
-          if [ -n "$state_file_status" ]; then
-            echo "TF_STATE_EXISTS=true" >> $GITHUB_ENV
+          if [ ${{ github.event_name }} == 'workflow_dispatch' ]; then
+            PR_NUMBER=${{ github.event.inputs.pr_number }}
+          else
+            PR_NUMBER=${{ github.event.pull_request.number }}
           fi
+          echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_ENV
+          echo "ENVIRONMENT=pr-${PR_NUMBER})" >> $GITHUB_ENV
```

Comment:
`ENVIRONMENT` feels very generic for a env var name with chance of clashes, etc? How about something something like `AZURE_ENVIRONMENT` ?